tests/integration_test.rs

```diff
@@ -382,3 +383,129 @@ async fn test_oci_manifest_capability(
         .send(())
         .expect("cannot send shutdown signal");
 }
+
+#[rstest]
+#[case::container_image("ghcr.io/kubewarden/tests/policy-server:v1.13.0", 
+    policy_fetcher::oci_distribution::manifest::OciImageManifest {
+        schema_version:2,
+        media_type: Some("application/vnd.docker.distribution.manifest.v2+json".to_owned()),
+        annotations: None,
+        artifact_type:None,
+        config: OciDescriptor{
+            media_type:  "application/vnd.docker.container.image.v1+json".to_owned(),
+            size: 1592,
+            digest: "sha256:bc3511804cb29da6333f0187a333eba13a43a3a0a1737e9b50227a5cf057af74".to_owned(),
+            annotations: None,
+            urls: None,
+        },
+    layers: vec![]}, Some(serde_json::json!({
+    "User": "65533:65533",
+    "ExposedPorts": {
+      "3000/tcp": {}
+    },
+    "Env": [
+      "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+    ],
+    "Entrypoint": [
+      "/policy-server"
+    ],
+    "WorkingDir": "/"})))]
+#[case::policy(
+    "ghcr.io/kubewarden/tests/context-aware-test-policy:latest",
+    policy_fetcher::oci_distribution::manifest::OciImageManifest {
+        schema_version:2,
+        media_type: None, annotations: None, artifact_type:None,
+        config: OciDescriptor{
+            media_type:  "application/vnd.wasm.config.v1+json".to_owned(),
+            size: 2,
+            digest: "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a".to_owned(),
+            annotations: None, urls: None,
+        },
+    layers: vec![]},
+    None
+)]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_oci_manifest_and_config_capability(
+    #[case] policy_uri: &str,
+    #[case] expected_manifest: OciImageManifest,
+    #[case] expected_config: Option<serde_json::Value>,
+) {
+    let (callback_handler_shutdown_channel_tx, callback_handler_shutdown_channel_rx) =
+        oneshot::channel();
+    let callback_builder = CallbackHandlerBuilder::new(callback_handler_shutdown_channel_rx);
+    let mut callback_handler = callback_builder
+        .build()
+        .await
+        .expect("cannot build callback handler");
+    let callback_handler_channel = callback_handler.sender_channel();
+
+    tokio::spawn(async move {
+        callback_handler.loop_eval().await;
+    });
```

Comment:
This block of setup code seems to be a pattern in every test, maybe we could refactor it to the common module